sim/battle.ts

```diff
@@ -1908,6 +1904,20 @@ export class Battle {
 			subFormat.onTeamPreview?.call(this);
 		}
 
+		if (this.requestState !== 'teampreview' && this.ruleTable.pickedTeamSize) {
+			this.add('clearpoke');
+			for (const side of this.sides) {
+				for (const pokemon of side.pokemon) {
+					// Still need to hide these formes since they change on battle start
+					const details = pokemon.details.replace(', shiny', '')
+						.replace(/(Zacian|Zamazenta)(?!-Crowned)/g, '$1-*')
+						.replace(/(Xerneas)(-[a-zA-Z?-]+)?/g, '$1-*');
+					this.addSplit(side.id, ['poke', pokemon.side.id, details, '']);
+				}
+			}
+			this.makeRequest('teampreview');
+		}
+
```

Comment:
Can this be refactored into a separate method? Then mods can just call it.

Reply:
@Karthik99999 are you willing to send a follow-up PR? The current state of this has a severe risk of bitrot in mods.

Reply:
What exactly is the issue for mods?